verifyFiles.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env python
+
+#Script to check whether the output files generated by HCP pipeline scripts are the same or not. It will check for the match between two files with respect the filename, size, content, modification time, distance etc.
+#Maintainters : Big Data Lab Team, Concordia University.
+#email : tristan.glatard@concordia.ca, laletscaria@yahoo.co.in
+
+import os
+import sys
+import subprocess
+import argparse
+import hashlib
+from collections import defaultdict,OrderedDict
+#study_folder_details_dict_list is a list for storing ordered dictionaries containing the details regarding the files of  individual subjects.
+study_folder_details_dict_list=[]
+#Method list_files_and_dirs is used for listing files and directories present in the input directory.
+#Input Argument : Directory Path
+def list_files_and_dirs(dirPath):
+        listFileAndDirs=[]
+	rootDir=dirPath
+        for dir_, _, files in os.walk(rootDir):
+           listFileAndDirs.append(os.path.relpath(dir_, rootDir))
+           for fileName in files:
+              relDir=os.path.relpath(dir_, rootDir)
+              relFile=os.path.join(relDir, fileName)
+              listFileAndDirs.append(relFile)
+	#Sort the files and subdirectories according to the modification time.
+        listFileAndDirs.sort(key=lambda x: os.path.getmtime(os.path.join(rootDir,x)))
+	return listFileAndDirs
+
+#Method populate_file_dir_dict is an ordered  python dictionary to save the status details of
+#each file and directory present in the listFileAndDirs list
+#Input parameter : List contianing the details of the path of each file and directory.
+def populate_file_dir_dict(listFileAndDirs,dirPath):
+        temp_dict=OrderedDict(defaultdict(list))
+        for relPath in listFileAndDirs:
+	   md5_digest=generate_checksum(dirPath,relPath)
+           dirDetails=os.stat(os.path.join(dirPath,relPath))
+	   temp_dict.setdefault(relPath, []).append(md5_digest)
+	   temp_dict.setdefault(relPath, []).append(dirDetails)
+        return temp_dict
```

Comment:
Also, the checksums should be computed in a second step, only for the common files, and only when file sizes match. Computing a checksum can be a very costly operation. See latest edits in https://docs.google.com/document/d/1qlJq5X4P95qXfzC1XpLi1g3zjrbNaAtxo7EMGWH0Ois/edit